### PR TITLE
Profile Photo Feature

### PR DESCRIPTION
**Pull request by dreymoreau**
Hello! here is what Danny, Dale, Sami and I have so far in terms with implementing profile pictures for the volunteer section on the website!! We wanted to hear feedback and thoughts about best practices or if we are on the right track? Thank you! :)

### CODE REVIEW

**Comment by MichalSvatos** [comment]
Linking any kind of content directly from 3rd party website is a bad idea and security risk. For the visitiors and the website itself. You have no control over what content is served through that link.

Yes, some links are simple like `website.com/avatar.jpg` , but some (like gravatar) looks like this `https://1.gravatar.com/avatar/3390dda3df30e78f2b4944d501c9696c0a386948b656df49675c7309544a94fa?size=128` - you have no idea if that's an image or a malicious script. It could be image now, but the source can be changed later without you knowing it.

(Also people could provide a link to 30x30 JPG or 2500x3000 6MB PNG.)

Probably there is a way to somehow sanitize the content, but still potential risk.
I would not recommend that.

**Comment by evert** [comment]
Gravatar is a widely accepted place to host avatars. Ran by WordPress and used by stackoverflow and GitHub. Seems like a very extreme take especially given that these are volunteer avatars

**Comment by MichalSvatos** [comment]
Gravatar was just an example. What if someone submit an address `myportfolio.net/thisisdefinitelyanimageandnotascript?sizewinkwink=666`? 

I'm just saying that if it were up to me I would not do that.

**Comment by evert** [comment]
Hey @MichalSvatos , these are not actually viable things you can do with an image tag. Definitely good to be vigilant though, but I don't think there's a real risk here.

**Comment by kieran** [comment]
Some thoughts:

I'd really love to see gravatars implemented instead of including static photos in this repo - this allows people to control their own images without our intervention. Most(?) tech people probably already use them for github

It might be nice to have multiple URLs per user, like:
```
{
  name: '...',
  twitter: '...',
  linkedIn: '...',
  etc...
}
```
and then display a linked icon for each (bonus points for including `aria-label` attributes for a11y)

I'd love to see the CSS prio mobile screens first (smaller cards, single column), then the experience can be enhanced for larger breakpoints (larger card? multiple columns?) like desktop computers

There appears to be a merge conflict preventing deploy previews atm - this will need fixing. It's not always easy if you haven't done it before, so do reach out if you want a co-pilot

**Comment by MichalSvatos** [comment]
## Gravatar
Sure, [there is an API](https://docs.gravatar.com/general/images/). Anyway it should be opt-in and there has to be fallback for others who don't want to or don't use gravatar at all.

## Multiple URLs
We can write anything to the JSON and then it's "just" about the design.

There are some questions we should answer:
- how many links, is there a limit?
- which services icons we will support? (or we'll link some library with all the icons?)
- do we really want to substitute services like [linktr.ee](https://linktr.ee/) or [omg.lol](https://home.omg.lol/)?

## CSS
I don't know if you're speaking about something specific, but my CSS was written mobile first. The only difference that there is no media query, because I experimented with `grid` and `minmax()` function.

**Comment by usrrname** [comment]
@dreymoreau thanks for updating us on this! I see this was left in draft mode which might explain the comments exploring alternate approaches. You could mark it "Ready for review" if you're past the WIP stage... 

It would be awesome if you could pair with @kieran to implement some of these changes, otherwise i'm psyched this page is coming together, and we discussed an initial descoped version to get it working and improve it later. I might add that there is a barrier to entry with mainenance of coffeescript and ~3 volunteers worked on this... I don't see static links as a dealbreaker in this version.

File: pages/volunteers-old.md
Comment (usrrname) [inline pages/volunteers-old.md:1]: Don't worry about "archiving" the previous volunteer page. We can always go back in Git history to restore that :D 

I would remove `pages/volunteers-old.md`

File: pages/volunteers-old.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-## Volunteers 
+## Profile Photos
```
Comment (usrrname) [inline pages/volunteers-old.md:1]:
```suggestion
## Volunteers
```

Recommend keeping the title consistent with the subject of the page. Ideal for web crawlers.

File: screens/volunteers/volunteers.json [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+                "profileURL": "https://twitter.com/mihok",
+                "profileURLText": "twitter.com",
+                "avatarName": "mattMihok"
+            },
+            {
+                "name": "Jen Chan",
+                "profileURL": "https://www.jenchan.biz/",
+                "profileURLText": "jenchan.biz",
+                "avatarName": "jen"
+            }
```
Comment (usrrname) [inline screens/volunteers/volunteers.json:109]: Factual edit: you can remove me from this as I'm stepping down

File: screens/volunteers/volunteers.json [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+                "profileURL": "https://5hel2l2y.github.io/",
+                "profileURLText": "5hel2l2y.github.io",
+                "avatarName": "sherry"
+            },
+            {
+                "name": "Erik Derohanian",
+                "profileURL": "https://www.linkedin.com/in/erikjd/",
+                "profileURLText": "linkedin.com",
+                "avatarName": "torontojs"
+            }
```
Comment (usrrname) [inline screens/volunteers/volunteers.json:39]: Factual edit: Erik is part of promo and moderation :) just gotta move his object into that array

**Comment by dreymoreau** [comment]
hi hi @usrrname coming back around this, been a hectic last week and a bit! hoping to pair and implement some changes with the group this week! :))) will keep you updated with how it all goes!

**Comment by evert** [comment]
Hey! just in case my original comment wasn't clear. I just asked for gravatars as a suggestion. This is not a blocker of course, and even if people are interested in doing this, this is the type of stuff that can be done over several PRs and doesn't have to be done now.

**Pull request closed without merging**